pkg/controllers/networking/node_controller.go

```diff
@@ -69,12 +69,22 @@ func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	var underlayAttached, overlayAttached bool
-	if underlayAttached, overlayAttached, err = utils.DetectNetworkAttachmentOfNode(r, node); err != nil {
+	var underlayAttached, overlayAttached, bgpAttached bool
+	if underlayAttached, overlayAttached, bgpAttached, err = utils.DetectNetworkAttachmentOfNode(r, node); err != nil {
 		log.Error(err, "unable to detect network attachment")
 		return ctrl.Result{}, err
 	}
 
+	// if global bgp network not exist, do not patch bgp attachment label to node
+	if bgpAttached {
+		if globalBGPNetwork, err := utils.FindGlobalBGPNetwork(r); err != nil {
+			log.Error(err, "unable to detect global bgp network exist")
+			return ctrl.Result{}, err
+		} else if len(globalBGPNetwork) == 0 {
+			bgpAttached = false
```

Comment:
if global bgp network not exists, I believe that not a bgp attachment label should be patched on nodes.

Reply:
It already is.

Reply:
I mean the bgp attachment label should not exist, but not be set to false.